src/tests/test_gcms_metadata_gen.py

```diff
@@ -27,3 +32,32 @@ def test_gcms_metadata_gen():
     # Run the metadata generation process
     generator.run()
     assert os.path.exists(output_file)
+
+
+def test_gcms_biosample_gen():
```

Comment:
Could you please add a test and exmaple data for the scenario where you have a calibration_id and don't generate calibration instances?  Thanks!